unlock-app/src/components/creator/CreatorLocks.js

```diff
@@ -43,9 +43,9 @@ export class CreatorLocks extends React.Component {
         </LockHeaderRow>
         <Errors />
         {showDashboardForm && <CreatorLockForm hideAction={this.toggleForm} />}
-        {lockFeed.map(lock => {
-          return <CreatorLock key={JSON.stringify(lock)} lock={lock} />
-        })}
+        {lockFeed.map(lock => (
+          <CreatorLock key={lock.address} lock={lock} edit={() => {}} />
```

Comment:
So, this is interesting. We previously had this `key={lock.address}` but apparently React uses the keys to identify if the component needs to be re-rendered. However it only checks using shallow verification which means that the component was actually not re-rendered even though some of its properties have changed. This is the reason I used `JSON.stringify(lock)`.

Reply:
ah. Another way to solve this is to do {...lock} instead of lock={lock}, then it will re-render on any change to the lock, but that is a future PR ;)